### PR TITLE
feat: remember the network and broadcast address

### PR DESCRIPTION
**Pull request by cloudcosmonaut**
I noticed that the count function didn't consider that a subnet requires a network and broadcast address. I also added the edge case of a /31 subnet and how that is handled by providers (even though it's not compliant ;))

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -33,23 +33,30 @@ This also works with IPv6 addresses, for example:
...
 
 To get all distinct host addresses part of a given CIDR range:
 
 ```
 $ cidr count 10.0.0.0/16
-65536
+65534
+```
+
+Or with an edge case:
```
[the remark this answers is not in the document]
Comment (bschaatsbergen) [reply README.md:47]: My suggested change is already a bit too too long, I would love a simplified version of my suggestion, thoughts?

**Review by bschaatsbergen — COMMENTED** [review]
Overall the contribution looks great, thanks for that :purple_heart:. Before approving I would like to get some clarity on the questions I've raised.

File: pkg/core/core.go [hunk 2]
```diff
@@ -6,7 +6,18 @@ import (
 
 func AddressCount(network *net.IPNet) uint64 {
 	prefixLen, bits := network.Mask.Size()
-	return 1 << (uint64(bits) - uint64(prefixLen))
+
+	// Check if network is IPv4 or IPv6
+	if network.Mask != nil {
+		// Handle edge cases
+		switch prefixLen {
+			case 32: return 0
```
Comment (bschaatsbergen) [inline pkg/core/core.go:14]:
```suggestion
			case 32: return 1
```

File: README.md [hunk 3]
```diff
@@ -33,23 +33,30 @@ This also works with IPv6 addresses, for example:
...
+
+```
+$ cidr count count 172.16.18.0/31
+1
```
[the remark this answers is not in the document]
Comment (cloudcosmonaut) [reply README.md]: You're right, I updated the code!

File: README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (bschaatsbergen) [reply README.md:47]:
```suggestion
Or with large prefixes like point-to-point links:
```

**Comment by bschaatsbergen** [comment]
Thanks for your contribution @cloudcosmonaut :purple_heart:

**Pull request merged**